### ISSUE TITLE
CA2000 not reported for types implementing IDisposable. It's reported only for IAsyncDisposable

### ISSUE THREAD

**Issue by cezarypiatek** [dotnet/roslyn-analyzers#4453]
### Analyzer

**Diagnostic ID**: [CA2000](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA2000): `Dispose objects before losing scope`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.1 (Latest)

### Describe the bug

CA2000 is correctly reported for `IAsyncDisposable` but I don't get any errors for `IDisposable`

### Steps To Reproduce

Example code

```
using System;
using System.IO;
using System.Threading.Tasks;

namespace DisposableExamples
{
    class Program
    {
        static void Main(string[] args)
        {
            var a = CreateAsyncDisposable(); //CA2000 reported
            var b = CreateDisposable(); //CA2000 not reported!!!
            var c = CreateOwnDisposable();  //CA2000 not reported!!!
            var d = File.Open("text.txt",FileMode.OpenOrCreate); //CA2000 reported
        }

        private static IDisposable CreateDisposable() => throw new NotImplementedException();
        private static IAsyncDisposable CreateAsyncDisposable() => throw new NotImplementedException();
        private static SampleDisposable CreateOwnDisposable() => throw new NotImplementedException();
    }

    class SampleDisposable: IDisposable, IAsyncDisposable
    {
        public void Dispose()
        {
        }

        public ValueTask DisposeAsync()
        {
            throw new NotImplementedException();
        }
    }
}
```

### Expected behavior
I'm expecting to get `CA2000` reported for lines with variables a,b,c,d

### Actual behavior
I'm getting `CA2000` for `a` and `d`. Moreover, when I remove `IDisposable` from the `SampleDisposable` then I'm getting `CA2000` for `c`. It looks like the analyzer is not working well with   `IDisposable`. Sample project is `netcoreapp3.1` console app.

**Comment by Evangelink** [comment]
I can confirm the behavior is still present on master. 

I will need to have a deeper look but from what I have seen the `CreateDisposable` and `CreateOwnDisposable` have no associated location while `CreateAsyncDisposable` and `File.Open` do. See the location returned here:
https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeDataFlowOperationVisitor.cs#L296

Tagging @mavasani in case you can directly see what's wrong.

**Comment by cytoph** [comment]
I have the exact same issue. Is there anything that'll be done to fix this? There's nothing new in this issue since January 2021!
It is really annoying having to always surround respective lines with pragma.
